### PR TITLE
Fix email parsing issue in text widget

### LINKED ISSUE #4101: [Bug] Text parsing issue
## Description
c.lala@something.com is rendered as `{{{url0}}}c.lala@something.com{{{/url0}}}` in the text widget.
email-binding

### Steps to reproduce the behaviour:
Add steps to reproduce this behaviour, include console / network logs & screenshots
1. Create a text widget
2. Add c.lala@something.com in the default text
3. See error

### Important Details
 Release 1.4.10

### PR DESCRIPTION
**Pull request by vicky-primathon**
## Description
Fix email parsing issue in text widget

Fixes #4101 

## Type of change
- Bug fix (non-breaking change which fixes an issue)

## How Has This Been Tested?
- Added cypress test to validate email parsing

## Checklist:

- [x] My code follows the style guidelines of this project
- [x] I have performed a self-review of my own code
- [x] I have commented my code, particularly in hard-to-understand areas
- [x] I have made corresponding changes to the documentation
- [x] My changes generate no new warnings
- [x] I have added tests that prove my fix is effective or that my feature works
- [x] New and existing unit tests pass locally with my changes

## Test coverage results :test_tube:
<details><summary>:white_circle: Total coverage has not changed</summary>
No changes to code coverage between the base branch and the head branch</details>

### CODE REVIEW

**Review by somangshu — DISMISSED** [review]
Approving for test run

**Comment by somangshu** [comment]
Tested on deploy preview.

File: app/client/src/components/designSystems/blueprint/EmailMatcher.tsx [hunk 1]
```diff
@@ -0,0 +1,81 @@
...
+
+interface CombinePatternsOptions {
+  capture?: boolean;
+  flags?: string;
+  join?: string;
+  match?: string;
+  nonCapture?: boolean;
+}
+
+function combinePatterns(
```
Comment (riodeuno) [inline app/client/src/components/designSystems/blueprint/EmailMatcher.tsx:15]: Let's add unit tests for this file.

File: app/client/src/components/designSystems/blueprint/TextComponent.tsx [hunk 2]
```diff
@@ -83,6 +84,10 @@ class TextComponent extends React.Component<TextComponentProps> {
       textColor,
       backgroundColor,
     } = this.props;
+    let matchers = [new UrlMatcher("url"), new EmailMatcher("email")];
+    if (text && EMAIL_PATTERN.test(text)) {
```
Comment (riodeuno) [inline app/client/src/components/designSystems/blueprint/TextComponent.tsx:88]: What happens when the content has both URL and emails?

**Review by riodeuno — CHANGES_REQUESTED** [review]

File: app/client/src/components/designSystems/blueprint/TextComponent.tsx [hunk 3]
```diff
@@ -3,7 +3,7 @@ import { Text } from "@blueprintjs/core";
 import styled from "styled-components";
 import { ComponentProps } from "components/designSystems/appsmith/BaseComponent";
 import { TextAlign } from "widgets/TextWidget";
-import Interweave from "interweave";
+import Interweave, { Node } from "interweave";
```
Comment (riodeuno) [inline app/client/src/components/designSystems/blueprint/TextComponent.tsx:6]: I don't see the use of `Node` which is imported here.

**Review by riodeuno — CHANGES_REQUESTED** [review]

File: app/client/src/components/designSystems/blueprint/TextComponent.tsx (hunk 3 above)
Comment (riodeuno) [reply app/client/src/components/designSystems/blueprint/TextComponent.tsx:6]: Also, the deploy preview failed to build. can we see what went wrong?

**Review by riodeuno — APPROVED** [review]
Looks good to me.

**Pull request merged**